textplot/utils.py

```diff
@@ -3,13 +3,83 @@
 import re
 import numpy as np
 import functools
+import copy
 
 from collections import OrderedDict
 from nltk.stem import PorterStemmer
 from itertools import islice
 
+import spacy
+from nltk.stem.snowball import SnowballStemmer
 
-def tokenize(text):
+class SpacyTokenizer(object):
+    def __init__(self, lang='de', lower=False):
+
+        if lang == 'de':
+            spacy_model = 'de_core_news_sm'
+        elif lang == 'en':
+            spacy_model = 'en_core_web_sm'
+        else:
+            print(f'Selected language not yet supported: {lang}')
+            raise NotImplementedError
+        print(f'Initializing tokenizer: Spacy tokenizer with model {spacy_model}')
+        self.nlp = spacy.load(spacy_model)
```

Comment:
for optimisation, I think there are some speedups available by specifying any unnecessary pipeline components (see [here](https://spacy.io/usage/processing-pipelines))

Reply:
We could also consider some other tools for faster/lightweight tokenization:

- https://github.com/microsoft/BlingFire
- https://pypi.org/project/sacremoses/